kernel_gateway/gatewayapp.py

```diff
@@ -528,5 +530,8 @@ def shutdown(self):
         """Stop all kernels in the pool."""
         self.personality.shutdown()
 
+    def _signal_stop(self, sig, frame):
+        self.log.info("Received signal to terminate.")
+        ioloop.IOLoop.current().stop()
```

Comment:
self.io_loop makes sense. `IOLoop.current()` uses thread-local storage, so may not find the right IO Loop, depending on whether the application and the event handler are running in the same thread.